Sources/Utilities/AudioVideoPlayer/AudioVideoPlayer.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  AudioVideoPlayer.swift
+//  PovioKit
+//
+//  Created by Toni Kocjan on 29/10/2021.
+//  Copyright © 2021 Povio Labs. All rights reserved.
+//
+
+import AVKit
+
+public protocol AudioVideoPlayerDelegate: AnyObject {
+  func audioVideoPlayer(didBeginPlayback player: AudioVideoPlayer)
+  func audioVideoPlayer(didEndPlayback player: AudioVideoPlayer)
+  func audioVideoPlayer(didPausePlayback player: AudioVideoPlayer)
+  func audioVideoPlayer(didBeginReplay player: AudioVideoPlayer)
+  func audioVideoPlayer(didBeginBuffering player: AudioVideoPlayer)
+  func audioVideoPlayer(didEndBuffering player: AudioVideoPlayer)
+  func audioVideoPlayer(_ player: AudioVideoPlayer, didProgressToTime seconds: Double)
+}
+
+public extension AudioVideoPlayerDelegate {
+  func audioVideoPlayer(didBeginPlayback player: AudioVideoPlayer) {}
+  func audioVideoPlayer(didEndPlayback player: AudioVideoPlayer) {}
+  func audioVideoPlayer(didPausePlayback player: AudioVideoPlayer) {}
+  func audioVideoPlayer(didBeginReplay player: AudioVideoPlayer) {}
+  func audioVideoPlayer(didBeginBuffering player: AudioVideoPlayer) {}
+  func audioVideoPlayer(didEndBuffering player: AudioVideoPlayer) {}
+  func audioVideoPlayer(_ player: AudioVideoPlayer, didProgressToTime seconds: Double) {}
+}
+
+public class AudioVideoPlayer: AVPlayer {
+  public static let defaultTimescale = CMTimeScale(kCMTimeMaxTimescale)
+
+  public private(set) lazy var playbackInterval: (startAt: Double, endAt: Double) = (0, duration)
+  public private(set) var isPlaying = false
+  public var allowLooping = true
+  public weak var delegate: AudioVideoPlayerDelegate?
+  
+  private var timeObserver: Any?
+
+  public override init() {
+    super.init()
+    setupTimeObserver()
+  }
+
+  public override init(url: URL) {
+    super.init(url: url)
+    setupTimeObserver()
+  }
+
+  override init(playerItem item: AVPlayerItem?) {
+    super.init(playerItem: item)
+    setupTimeObserver()
+  }
+  
+  convenience public init?(urlString: String) {
+    guard let url = URL(string: urlString) else { return nil }
+    self.init(url: url)
+  }
+}
+
+// MARK: - API
+public extension AudioVideoPlayer {
+  override func play() {
+    super.play()
+    isPlaying = true
```

Comment:
Do we really need this flag? I would rather go with [rate](https://developer.apple.com/documentation/avfoundation/avplayer/1388846-rate?preferredLanguage=occ) instead. What do you think?
```swift
var isPlaying: Bool {
  rate != 0
}
```

Reply:
This looks very promising, yes. 👍  Will test it out if it works as expected.